agent/run_job.go

```diff
@@ -148,6 +157,50 @@ func (r *JobRunner) Run(ctx context.Context) error {
 	return nil
 }
 
+func (r *JobRunner) prependTimestampForLogs(s string, args ...any) []byte {
```

Comment:
right now this func both prepending stuff and also sprintf-ing its args... is there a reason that it has to happen all in one go? or can we sprintf prior to passing something to this func?

Reply:
It just seems easier to call one method rather than 2. If you want to sprintf prior to this method, then this interface does not prevent that. But if you don't want to, the suggested interface does.

But I should rename the method, its name only suggests the prepending, not the sprintfing.